src/main.cpp

```diff
@@ -2717,19 +2717,14 @@ bool static ConnectTip(CValidationState& state, const CChainParams& chainparams,
     int64_t nTime5 = GetTimeMicros(); nTimeChainState += nTime5 - nTime4;
     LogPrint("bench", "  - Writing chainstate: %.2fms [%.2fs]\n", (nTime5 - nTime4) * 0.001, nTimeChainState * 0.000001);
     // Remove conflicting transactions from the mempool.
-    list<CTransaction> txConflicted;
-    mempool.removeForBlock(pblock->vtx, pindexNew->nHeight, txConflicted, !IsInitialBlockDownload());
+    list<CTransaction> txConflictedCurrent;
+    mempool.removeForBlock(pblock->vtx, pindexNew->nHeight, txConflictedCurrent, !IsInitialBlockDownload());
+    txConflicted.insert(txConflicted.end(), txConflictedCurrent.begin(), txConflictedCurrent.end());
```

Comment:
Without a temporary variable, `mempool.removeForBlock` would always work though the combined list `txConflicted` that was initialized [here](https://github.com/bitcoin/bitcoin/pull/7946/files/9471f32a8fad340d0518ded6320fc1402f1d6286#diff-7ec3c68a81efff79b6ca22ac1f1eabbaR2917).

Although I'm not sure if this would be a problem, but at least it would slightly change currents master behavior.